src/main/ruby/core/hash.rb

```diff
@@ -183,6 +183,24 @@ def assoc(key)
     nil
   end
 
+  def compact
+    return Hash.new if empty?
```

Comment:
`return {} if empty?` seems simpler here.
But this shortcut should not be needed, `reject` can handle empty Hash instances.